src/NServiceBus.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -165,6 +167,30 @@ void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
 
             try
             {
+                // The current thread will be the event-raising thread if either:
+                //
+                // a) the semaphore was entered synchronously (did not have to wait).
+                // b) the event was raised on a thread pool thread,
+                //    and the semaphore was entered asynchronously (had to wait),
+                //    and the continuation happened to be scheduled back onto the same thread.
+                if (Thread.CurrentThread.ManagedThreadId == eventRaisingThreadId)
+                {
+                    // In RabbitMQ.Client 4.1.0, the event is raised by reusing a single explicitly created thread
```

Comment:
Oh, and I think there should probably be one between "single" and "explicitly" because those are both modifying "thread" and could be applied separately.